packages/datagateway-common/src/views/viewButton.component.test.tsx

```diff
@@ -69,47 +65,32 @@ describe('Generic view button', () => {
     handleButtonChange.mockClear();
   });
 
-  it('renders correctly', () => {
-    const wrapper = createRTLWrapper(props);
-    expect(wrapper.asFragment()).toMatchSnapshot();
+  it('renders correctly', async () => {
+    const { asFragment } = renderComponent(props);
+    expect(asFragment()).toMatchSnapshot();
   });
 
-  it('calls the handle button change when the view button is clicked', () => {
-    const wrapper = createWrapper(props);
-
-    wrapper
-      .find('[aria-label="page view app.view_table"]')
-      .last()
-      .simulate('click');
-
+  it('calls the handle button change when the view button is clicked', async () => {
+    renderComponent(props);
+    await user.click(
+      await screen.findByRole('button', {
+        name: 'page view app.view_table',
+      })
+    );
     expect(handleButtonChange).toHaveBeenCalledTimes(1);
-
-    wrapper.update();
-
-    expect(
-      wrapper.find('[aria-label="page view app.view_cards"]')
-    ).toBeTruthy();
-
-    wrapper
-      .find('[aria-label="page view app.view_table"]')
-      .last()
-      .simulate('click');
-
-    expect(handleButtonChange).toHaveBeenCalledTimes(2);
-
-    expect(
-      wrapper.find('[aria-label="page view app.view_cards"]')
-    ).toBeTruthy();
   });
```

Comment:
Is it possible to add back in the testing statements that it switches state to then be a `page view app.view_cards` button that when called switches it back to the first button?